avro-fastserde/src/test/java/com/linkedin/avro/fastserde/FastSpecificDeserializerGeneratorTest.java

```diff
@@ -824,6 +824,23 @@ public void shouldReadUnionOfRecordFieldsWithSameEnumFieldName(Boolean whetherUs
     Assert.assertEquals(getField((MyRecordV2) getField(deserRecord, "unionField"), "enumField"), MyEnumV2.VAL3);
   }
 
+  @Test(groups = {"deserializationTest"})
```

Comment:
This new test should generate some test classes when executing the test case, so can you also check in the generated test classes?

Normally, we will divide the commit into two:
1. Main code change including test cases.
2. Generated codes.

An example:
https://github.com/linkedin/avro-util/pull/326/commits